src/main/java/org/folio/auth/authtokenmodule/storage/TokenStore.java

```diff
@@ -0,0 +1,79 @@
+package org.folio.auth.authtokenmodule.storage;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowSet;
+import io.vertx.sqlclient.SqlConnection;
+import io.vertx.sqlclient.Tuple;
+
+import org.folio.tlib.postgres.TenantPgPool;
+
+/**
+ * The base class for all token storage. Actual token store classes should
+ * extend this.
+ */
+public class TokenStore {
+  private static final Logger log = LogManager.getLogger(TokenStore.class);
+
+  protected Vertx vertx;
+  protected String tenant;
+  protected TenantPgPool pool;
+
+  public TokenStore(Vertx vertx, String tenant) {
+    this.vertx = vertx;
+    this.tenant = tenant;
+    this.pool = TenantPgPool.pool(vertx, tenant);
+  }
+
+  /**
+   * Removes all tokens from storage for the tenant associated with this instance
+   * of the token store.
+   * @param conn An SqlConnection object to be used by the method to access the token
+   * store. It is the responsibility of callers to close this connection.
+   * @param suffix The string appended to the end of the table name which identifies
+   * the table.
+   * @return A succeeded future should the operation succeed, otherwise a failed
+   * future.
+   */
+  public Future<Void> removeAll(SqlConnection conn, String suffix) {
+    log.info("Removing all tokens from storage");
+
+    String delete = "DELETE FROM " + tableName(tenant, suffix);
+
+    return conn.preparedQuery(delete).execute().mapEmpty();
+  }
+
+  protected Future<Row> getRow(SqlConnection conn, String select, Tuple where) {
```

Comment:
All calls to this method can be changed to call `getRow(String select, Tuple where)` so that this method can call `pool.preparedQuery(select)`

Reply:
Please check whether `SqlConnection` parameter of the other methods of TokenStore can also been removed.

Reply:
> All calls to this method can be changed to call `getRow(String select, Tuple where)` so that this method can call `pool.preparedQuery(select)`

I think with `getRow` I want to keep passing in the `SqlConnection`, because `RefreshTokenStore.checkTokenNotRevoked` uses it and should share the same connection for its operations so using `withConnection` makes good sense there.

But I think you are right that the other methods in `TokenStore`. They don't need this parameter and can use the pool directly.

Reply:
How about
```
  protected Future<Row> getRow(String select, Tuple where) {
    return pool.withConnection(conn -> getRow(conn, select, where));
  }
```